packages/permissions/src/hooks/use-is-authorized/use-is-authorized.ts

```diff
@@ -22,7 +19,7 @@ const useIsAuthorized = ({
       `React hooks do not seem to be available yet in the installed React version "${React.version}". Please check the React Hooks documentation for more info: https://reactjs.org/hooks.`
     );
 
-  const actualPermissions = useApplicationContext<SelectedApplicationContext>(
+  const actualPermissions = useApplicationContext<TPermissions>(
```

Comment:
Won't do the trick. Not sure why.

Reply:
I think because the generic type `SelectedContext extends {} = {}` in `useApplicationContext` is wrong, it shouldn't be only an object.

I'll open a separate PR to fix it

Reply:
Hmm actually the signature `useApplicationContext` might not be the problem. For example, if I pass `string` but return a `number`, the compiler errors:

<img width="855" alt="image" src="https://user-images.githubusercontent.com/1110551/59456600-fad1f800-8e16-11e9-8070-2492663011b8.png">

Reply:
So I made some fixes/changes to the app context types (#740). With those changes, we now get the following:

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/1110551/59463411-358f5c80-8e26-11e9-844c-f339346f9a9c.png">

You can also see how function overloading works in this case:

<img width="889" alt="image" src="https://user-images.githubusercontent.com/1110551/59463452-4b9d1d00-8e26-11e9-875a-1ec160ad82bc.png">